### PR TITLE
Fix Unit Tests to Run Serially

### PR DESCRIPTION
**Pull request by mitchute**
Fixes some unit tests that don't pass when tests executed serially.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/DataWater.cc [hunk 1]
```diff
@@ -122,6 +122,11 @@ namespace DataWater {
     bool WaterSystemGetInputCalled(false); // set true once input data gotten.
     bool AnyIrrigationInModel(false);      // control flag set true if irrigation input for ecoroof DJS PSU Dec 2006
 
+    void clear_state()
+    {
+        WaterSystemGetInputCalled = false;
```
Comment (mjwitte) [inline src/EnergyPlus/DataWater.cc:131]: This is a good start, but the clear_state should reset all of the non-const variables above.

**Review by mjwitte — COMMENTED** [review]
Testing here. Confirmed that develop fails. 

We should be catching these during review (there *is* a checklist item for "Run Unit Test(s) locally". Maybe we should add "serially".) It can take a while for them all to run, but I exclude the property data tests 'cause they're *reallly* slow.
`energyplus_tests.exe --gtest_filter=-*PropertiesRefData*`

**Comment by mjwitte** [comment]
Also, running all tests on this branch, I got this array bounds failure:
```
EnergyPlusFixture.WindowEquivalentLayer_VBMaximizeBeamSolar
Assertion failed: contains( i ), file C:\MJW Git\EnergyPlus\third_party\ObjexxFCL\src\ObjexxFCL/Array1.hh, line 1172
```

**Comment by mitchute** [comment]
@mjwitte @Myoldmopar this now passes locally in debug and release modes, but I see yours is still failing. I'll take a look. And yes, we should be adding all of the non-const variables to clear_states. The globals refactor work will eventually get this done, but I'll add the ones in the files affected here.

E+ Dev Team: if anyone sees non-consts that should be added to clear_states, feel free to add them. Some namespaces (like in this fix) didn't even have a clear_state, so if you find that feel free to add one. Last year I was playing around with the clang AST matcher tools and found something like 1200 ```static``` variables, and 13,500 variables with global scope. The summary is here, though, it's quickly becoming out of date: https://docs.google.com/spreadsheets/d/1wSEx6BbjdW3IhxOqb_SLOg29W04QY9Lf99eh5RH0GBc/edit?usp=sharing As you work, feel free to eliminate as many of these as you can.

**Comment by mitchute** [comment]
> Hmm, will that do the same as `RainFaill.deallocate();` as other `clear_states` do? I was expecting to see `NumWaterStorageTanks = 0;` etc. here as well.

```RainFall``` is a just an struct instance, not an array or vector, so we can't use ```deallocate```. The ```MyStructInst = {};``` seems to be an accepted C++ 11 way of resetting a class instance (https://stackoverflow.com/a/15183438)

**Comment by mitchute** [comment]
> Also, running all tests on this branch, I got this array bounds failure:
> 
> ```
> EnergyPlusFixture.WindowEquivalentLayer_VBMaximizeBeamSolar
> Assertion failed: contains( i ), file C:\MJW Git\EnergyPlus\third_party\ObjexxFCL\src\ObjexxFCL/Array1.hh, line 1172
> ```

@mjwitte did you run this with the latest commit (c0f4b78b25cf4be2a274c88dec632960d9ea2a3c). I was hitting that error also before this latest commit.

**Comment by mjwitte** [comment]
Hmm, maybe not. I'll pull again and re-run.
Regarding `.deallocate()` I was moving too quickly.  The `Array1D`s below the struct should be deallocated, like `WaterStorage` etc.?

**Comment by mitchute** [comment]
Yes, they should. I just wanted to make sure to identify the ones that were causing failures before doing "extra".

**Comment by mjwitte** [comment]
Ok, sorry to jump the gun on reviewing. Holler when you're done.

**Comment by mjwitte** [comment]
@Myoldmopar If a PR is labeled `DoNotPublish` and `Defect` will it show up in the changelog?

**Comment by Myoldmopar** [comment]
> @Myoldmopar If a PR is labeled `DoNotPublish` and `Defect` will it show up in the changelog?

It will [not](https://github.com/NREL/EnergyPlus/blob/develop/doc/tools/create_changelog.py#L127-L130).  Only NewFeature, Defect, Performance, and Refactoring are dropped in the changelog.  The purpose of DoNotPublish is just to explicitly enforce that all PRs should be labeled with "something".

**Comment by mjwitte** [comment]
I'm confused, it looks like a PR with both labels `Defect` and `DoNotPublish` (like this one) *will* show up in the changelog. So, if you don't want something in the changelog, it should just have a single `DoNotPublish` label.

**Comment by Myoldmopar** [comment]
Great catch!  Yes, I was forgetting about PRs with multiple labels.  In a DoNotPublish case, you shouldn't put on another label, or it will be published.  I can change that behavior so that DoNotPublish overrides other PRs for the changelog.  I'll make that change soon, that way PRs can still use the other labels to reflect their intent.

**Comment by mitchute** [comment]
I'm done here unless something else breaks. Feel free to review/test.

**Review by mjwitte — APPROVED** [review]
All good here. Will merge once CI completes.

**Comment by mjwitte** [comment]
So, do we attribute this [Mac failure](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-174b39b1710514eff5c03139c6be129e615ac415-x86_64-MacOS-10.13-clang.html) to Mac randomness, or does it need further investigation? This seems to be a different failure that what we've been getting from Mac CI recently.

**Comment by mitchute** [comment]
I just ran the tests 3 times locally (on my Mac) and didn't see any troubles. I suppose I could clean up a few more variables and see is CI still complains. Thoughts?

**Comment by Myoldmopar** [comment]
I just found that that same error was in other branches, so I think this is OK...

**Pull request merged**